freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/PasswordService.java

```diff
@@ -121,6 +130,25 @@ private void asyncSetPasswords(Optional<String> requestId, String operationId,
         }
     }
 
+    @VisibleForTesting
+    Optional<Instant> calculateExpirationTime(String actorCrn, String accountId) {
+        LOGGER.debug("calculating expiration time for password in account {}", accountId);
+        UserManagementProto.Account account = umsClient.getAccountDetails(actorCrn, accountId, MDCUtils.getRequestId());
+        if (account.hasPasswordPolicy()) {
+            long maxLifetime = account.getPasswordPolicy().getWorkloadPasswordMaxLifetime();
+            if (maxLifetime != 0L) {
+                LOGGER.debug("Calculating password expiration by adding lifetime '{}' to current time", maxLifetime);
+                return Optional.of(Instant.ofEpochMilli(currentTimeSupplier.get() + maxLifetime));
```

Comment:
What is the reason for the supplier? you can use the _Clock_ class if you want to mock the current millis.

Reply:
yes it was just for testing.

Reply:
yes it was just for testing.

Reply:
changed to Clock